chats/apps/api/v1/rooms/viewsets.py

```diff
@@ -214,16 +213,47 @@ def perform_update(self, serializer):
         msg.notify_room("create")
 
         if old_user is None and user:  # queued > agent
-            instance.notify_queue("update", transferred_by=self.request.user.email)
+            instance.notify_queue("update")
         elif old_user is not None:
-            instance.notify_user(
-                "update", user=old_user, transferred_by=self.request.user.email
-            )
+            instance.notify_user("update", user=old_user)
             if queue:  # agent > queue
-                instance.notify_queue("update", transferred_by=self.request.user.email)
+                instance.notify_queue("update")
             else:  # agent > agent
-                instance.notify_user("update", transferred_by=self.request.user.email)
+                instance.notify_user("update")
 
     def perform_destroy(self, instance):
         instance.notify_room("destroy", callback=True)
         super().perform_destroy(instance)
+
+    @action(
+        detail=True,
+        methods=["PATCH"],
+    )
+    def update_custom_fields(self, request, pk=None):
+        custom_fields_update = request.data
+        data = {"fields": custom_fields_update}
```

Comment:
The term `data` is quite generic and can refer to many things. it's hard to tell what this field means without going into its declaration.

you could call it `custom_fields` for example and at the time of the request it would be: `data=custom_fields`